File: ss-front/src/app/model/Patient.ts
import {GeneralInfo} from "./GeneralInfo";
import {WishesMedication} from "./WishesMedication";
import {Photo} from "./Photo";
import {Scan} from "./Scan";

export interface Patient {
  id: number;
  generalInfoId: number;
  wishesMedicationId: number;
  photoIds: number[];
  scanIds: number[];
  generalInfo: GeneralInfo;
  wishesMedication: WishesMedication;
  photos: Photo[];
  scans: Scan[];
  addingDate: Date;
}

File: ss-back/src/main/java/stomat/ssback/model/Patient.java
package stomat.ssback.model;

import jakarta.persistence.*;
import lombok.*;
import stomat.ssback.model.medication.WishesMedication;
import stomat.ssback.model.photo.Photo;
import stomat.ssback.model.scan.Scan;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity(name = "patients")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    @OneToOne
    private GeneralInfo generalInfo;

    @OneToOne
    private WishesMedication wishesMedication;

    @Column(name = "photos")
    @OneToMany(cascade = CascadeType.REMOVE)
    private List<Photo> photos;

    @Column(name = "scans")
    @OneToMany
    private List<Scan> scans;

    @Column(name = "adding_date")
    private LocalDate addingDate;
}
